src/app/api/role/[_id]/route.ts

```diff
@@ -0,0 +1,80 @@
+import { NextRequest, NextResponse } from "next/server";
+import connectDB from "@database/db";
+import volunteerRoleSchema from "@database/volunteerRoleSchema";
+import Events from "@database/eventSchema";
+import VolunteerRoles from "@database/volunteerRoleSchema";
+
+type IParams = {
+  params: {
+    _id: string;
+  };
+};
+
+export async function GET(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params; //destructure
+  try {
+    const volunteerRole = await volunteerRoleSchema.findOne({ _id }).orFail();
+    return NextResponse.json(volunteerRole);
+  } catch (err) {
+    console.log(err);
+    return NextResponse.json("Could not find the Volunteer Role.", {
+      status: 404,
+    });
+  }
+}
+
+export async function DELETE(req: NextRequest, { params }: IParams) {
+  await connectDB();
+  const { _id } = params; //destructure
+
+  try {
+    const volunteerRole = await VolunteerRoles.findOne({ _id }).orFail();
+    const eventid = volunteerRole.event; //gets event field from volunteerRole
+
+    await Events.findByIdAndUpdate(
+      eventid,
+      { $pull: { roles: _id } }, //removes role from event array
+      { new: true }
+    );
+
+    const deletedVolunteerRole = await volunteerRoleSchema
+      .findOneAndDelete({ _id }) //deltes volunteer role
+      .orFail();
+    if (!deletedVolunteerRole) {
+      return NextResponse.json("Could not find the Volunteer Role.", {
+        status: 404,
+      });
+    }
+    return NextResponse.json(deletedVolunteerRole); //returns deleted volunteer role
+  } catch (err) {
+    console.log(err);
+    return NextResponse.json("An error occurred.", { status: 500 });
+  }
+}
+
+export async function PUT(req: NextRequest, { params }: IParams) {
```

Comment:
@hnadrian please let me know what you think here! The convention for REST PUTs is to replace the entire document. This code is more conventional of a PATCH (replace individual field/s). I don't see why we can't have both since this is still good code and useful, and the PUT should be much simpler, so I think we should have you guys change this to a PATCH (I think you can literally just change the name) and then add a PUT that simply replaces the entire document, which will be passed in the request body.

Reply:
After talking to Adrian, please just change this to a PATCH endpoint (just change the function name to PATCH I think) and we will make the PUT a separate issue in another sprint. Once you do this and test with postman/boomerang/etc. go ahead and commit and we will merge this issue! @gracebw7 @IvanA673